### DISCUSSION TITLE
Understanding Combobox with Tabs example

### DISCUSSION THREAD

**Discussion by jaknas** [ariakit/ariakit#3732] [Questions]
When reading the APG for `combobox`, it mentions that [dialogs do not support `aria-activedescendant` ](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#:~:text=Unlike%20other%20combobox%20popups%2C%20dialogs%20do%20not%20support%20aria%2Dactivedescendant%20so%20DOM%20focus%20moves%20into%20the%20dialog%20from%20the%20combobox.). ARIA explanation for `aria-activedescendant` also seems [to support that](https://w3c.github.io/aria/#aria-activedescendant:~:text=The%20element%20with%20DOM%20focus%20is%20a%20combobox%2C%20textbox%20or%20searchbox%20with%20aria%2Dcontrols%20referring%20to%20an%20element%20that%20supports%20aria%2Dactivedescendant%2C%20and%20the%20value%20of%20aria%2Dactivedescendant%20refers%20to%20an%20accessibility%20descendant%20of%20the%20controlled%20element.).

In the Combobox with Tabs example, the popover that's being rendered is of type `dialog`. This is used in conjunction with `aria-activedescendant` on the input (`combobox`). 

I'd like to understand how that works. Is it because the `aria-activedescendant` points to a descendant of a valid role, like `tablist` in this case, or `option` when it's inside `listbox`? Thus effectively "ignoring" the fact that everything is a descendant of a `role='dialog'`.

Hopefully my question is somewhat clear 🙏

**Comment by diegohaz** [comment]
It works because the DOM focus is on a `role=combobox` element, which supports `aria-activedescendant`. Whether the options are rendered within a `role=dialog` element is irrelevant, provided the dialog doesn't use `aria-modal="true"`—which would remove the focused combobox from the accessibility tree. This wouldn't work if the focus were elsewhere.

Does this answer your question?

**Reply by jaknas** [reply]
Let me try to rephrase.

The APG for `role=combobox` does specify all viable roles of a popup, those are `listbox`, `grid`, `tree`, `dialog`. For each of them, there's a note specifying the needed focus behavior. For example:

> DOM Focus is maintained on the combobox and the assistive technology focus is moved within the **tree**/**listbox**/**grid** using aria-activedescendant as described in [Managing Focus in Composites Using aria-activedescendant](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_focus_activedescendant).

However, for `role=dialog`, it specifies:
> Unlike other combobox popups, dialogs do not support aria-activedescendant so DOM focus moves into the dialog from the combobox.

Further down the page, it's confirmed that `aria-activedescendant` always refers to `role=combobox` having it set:
> When a descendant of a **listbox**, **grid**, or **tree** popup is focused, DOM focus remains on the combobox and the combobox has [aria-activedescendant](https://w3c.github.io/aria/#aria-activedescendant) set to a value that refers to the focused element within the popup.

I interpret it, as If I were not able to use `aria-activedescendant` on `role=combobox` in combination with `role=dialog` popup.
